shared/nav.desktop.js

```diff
@@ -64,51 +65,77 @@ class Nav extends Component {
   }
 
   _resizeWindow () {
```

Comment:
this thing was very fragile before and had a bunch of edge cases, this should be better